sdk/core/azure-core/src/uuid.cpp

```diff
@@ -45,24 +49,14 @@ namespace Azure { namespace Core {
   {
     uint8_t uuid[UuidSize] = {};
 
-#if defined(AZ_PLATFORM_WINDOWS)
-    std::random_device rd;
+    // RAND_MAX == INT_MAX
+    std::uniform_int_distribution<uint32_t> distribution(0, RAND_MAX);
```

Comment:
Why are we asking for uint32_t but then using INT_MAX instead? Should this be UINT_MAX?